### PR TITLE
Add E2E test to cover manual instrumentation aggregation

### PR DESCRIPTION
**Pull request by akopich**
The `orbit_manual_instrumentation` now checks for the
presence of the manual instrumentation scopes and for
correctness of their types.

Tests: E2E
Bug: http://b/232939690

### CODE REVIEW

File: contrib/automation_tests/orbit_instrument_function.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def main(argv):
         CheckTimers(track_name_filter="hello_ggp_stand", expect_exists=False),
         FilterAndHookFunction(function_search_string='DrawFrame'),
         Capture(),
-        VerifyFunctionCallCount(function_name='DrawFrame', min_calls=30, max_calls=3000),
+        VerifyScopeTypeAndCallCount(function_name='DrawFrame', min_calls=30, max_calls=3000),
```
Comment (dpallotti) [inline contrib/automation_tests/orbit_instrument_function.py:44]: The first parameter was renamed to `scope_name`.

File: contrib/automation_tests/orbit_load_presets_2.py [hunk 2]
```diff
@@ -44,8 +44,8 @@ def main(argv):
         VerifyFunctionHooked(function_search_string='GgpIssueFrameToken'),
         # Verify that the functions from the presets end up in the capture.
         Capture(),
-        VerifyFunctionCallCount(function_name='DrawFrame', min_calls=30, max_calls=3000),
-        VerifyFunctionCallCount(function_name='GgpIssueFrameToken', min_calls=30, max_calls=3000),
+        VerifyScopeTypeAndCallCount(function_name='DrawFrame', min_calls=30, max_calls=3000),
+        VerifyScopeTypeAndCallCount(function_name='GgpIssueFrameToken', min_calls=30, max_calls=3000),
```
Comment (dpallotti) [inline contrib/automation_tests/orbit_load_presets_2.py:48]: The first parameter was renamed to `scope_name`.

File: contrib/automation_tests/orbit_manual_instrumentation.py [hunk 3]
```diff
@@ -10,7 +10,8 @@
 from test_cases.connection_window import FilterAndSelectFirstProcess, ConnectToStadiaInstance
 from test_cases.capture_window import Capture, CheckTimers, VerifyTracksDoNotExist, VerifyTracksExist, ToggleCollapsedStateOfAllTracks, FilterTracks
 from test_cases.symbols_tab import LoadSymbols
-"""Test Orbit's manual instrumentation.
+from test_cases.live_tab import VerifyScopeTypeAndCallCount
+"""Test Orbit"s manual instrumentation.
```
Comment (dpallotti) [inline contrib/automation_tests/orbit_manual_instrumentation.py:14]:
```suggestion
"""Test Orbit's manual instrumentation.
```

File: contrib/automation_tests/orbit_manual_instrumentation.py [hunk 4]
```diff
@@ -35,19 +36,31 @@ def main(argv):
         "float_var", "int64_var", "int_var", "uint64_var", "uint_var", "ORBIT_ASYNC_TASKS",
         "ORBIT_START_ASYNC_TEST"
     ]
+    scope_types = ["MA", "MS", "MS","MS","MS","MA","MS","MS","MS","MS","MS"]
```
Comment (dpallotti) [inline contrib/automation_tests/orbit_manual_instrumentation.py:39]: Spaces are inconsistent. Use `yapf`.

File: contrib/automation_tests/orbit_user_space_instrumentation.py [hunk 5]
```diff
@@ -35,7 +35,7 @@ def main(argv):
         LoadSymbols(module_search_string="hello_ggp"),
         FilterAndHookFunction(function_search_string='DrawFrame'),
         Capture(user_space_instrumentation=True),
-        VerifyFunctionCallCount(function_name='DrawFrame', min_calls=30, max_calls=3000),
+        VerifyScopeTypeAndCallCount(function_name='DrawFrame', min_calls=30, max_calls=3000),
```
Comment (dpallotti) [inline contrib/automation_tests/orbit_user_space_instrumentation.py:38]: The first parameter was renamed to `scope_name`.

File: contrib/automation_tests/orbit_user_space_instrumentation_silenus.py [hunk 6]
```diff
@@ -37,7 +37,7 @@ def main(argv):
         FilterAndHookFunction(function_search_string="Render"),
         Capture(user_space_instrumentation=True),
         CheckTimers(track_name_filter="triangle.exe", require_all=False),
-        VerifyFunctionCallCount(function_name="Render", min_calls=1000, max_calls=15000),
+        VerifyScopeTypeAndCallCount(function_name="Render", min_calls=1000, max_calls=15000),
```
Comment (dpallotti) [inline contrib/automation_tests/orbit_user_space_instrumentation_silenus.py:40]: The first parameter was renamed to `scope_name`.

File: contrib/automation_tests/test_cases/live_tab.py [hunk 7]
```diff
@@ -62,32 +63,36 @@ class AddFrameTrack(LiveTabTestCase):
...
 
-        logging.info('Found a call count of %s, hook status: %s', call_count, hook_status)
+        logging.info('Found a call count of %s, scope type: %s', call_count, actual_scope_type)
         self.expect_true(
             min_calls <= call_count <= max_calls,
-            'Call count for function "%s" expected to be between %s and %s, was %s' %
-            (function_name, min_calls, max_calls, call_count))
+            'Call count for scope "%s" expected to be between %s and %s, was %s' %
+            (scope_name, min_calls, max_calls, call_count))
+        self.expect_true(
```
[the remark this answers is not in the document]
Comment (akopich) [reply contrib/automation_tests/test_cases/live_tab.py:92]: `scope_type` isn't actually just the type, but also `H` and `F`. And I don't think we wish to always test for the correct `H` and `F`. So either we allow it to be `None`, or only check for the correct type, ignoring `H` and `F`.

If we wish to check for `scope_type` with `H` and `F` nevertheless, this should be done in another PR.

File: contrib/automation_tests/orbit_capture_loading.py [hunk 8]
```diff
@@ -74,7 +74,7 @@ def main(argv):
         VerifyTracksExist(track_names=["Page*", "*System*", "*CGroup*"], allow_duplicates=True),
 
         AddIterator(function_name="TestFunc2"),
-        VerifyFunctionCallCount(function_name="TestFunc2", min_calls=1257, max_calls=1257),
+        VerifyScopeTypeAndCallCount(function_name="TestFunc2", min_calls=1257, max_calls=1257),
```
[the remark this answers is not in the document]
Comment (akopich) [reply contrib/automation_tests/orbit_capture_loading.py:77]: Thank you!

File: contrib/automation_tests/test_cases/live_tab.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (dpallotti) [reply contrib/automation_tests/test_cases/live_tab.py:92]: What about just checking the prefix of `actual_scope_type`?

[the remark this answers is not in the document]
Comment (dpallotti) [reply contrib/automation_tests/test_cases/live_tab.py:92]: I mean, would that need to be another pull request?

File: contrib/automation_tests/orbit_capture_loading.py [hunk 9]
```diff
@@ -49,32 +48,27 @@ def main(argv):
...
-
         FilterTracks(filter_string=""),
         VerifyTracksExist(track_names=["Page*", "*System*", "*CGroup*"], allow_duplicates=True),
-
         AddIterator(function_name="TestFunc2"),
-        VerifyFunctionCallCount(function_name="TestFunc2", min_calls=1257, max_calls=1257),
+        VerifyScopeTypeAndHitCount(scope_name="TestFunc2",
+                                   scope_type="D",
+                                   min_calls=1257,
+                                   max_calls=1257),
```
Comment (dpallotti) [inline contrib/automation_tests/orbit_capture_loading.py:71]: Soo, same problem as before, you renamed these parameters. Same for other files.

File: contrib/automation_tests/test_cases/live_tab.py [hunk 10]
```diff
@@ -19,13 +19,14 @@ def __init__(self, **kwargs):
         self._live_tab = None
 
     def execute(self, suite):
+        self.find_control("TabItem", "Live", parent=suite.top_window()).click_input()
         self._live_tab = self.find_control("Group", "liveTab", parent=suite.top_window())
         super().execute(suite)
 
-    def find_function_cell(self, function_name) -> Tuple[BaseWrapper, int]:
+    def find_scope_cell(self, scope_name) -> Tuple[BaseWrapper, int]:
```
Comment (danielfenner) [inline contrib/automation_tests/test_cases/live_tab.py:26]: The overload of the term "scope" is a bit unfortunate here. It took several minutes for me to understand that a scope can be a function call or a manual instrumentation scope. A comment here and / or somewhere in `VerifyScopeTypeAndHitCount` might help.

Also: I'm not exactly a python wizard but I think this method should be named `_find_scope_cell` to encourage protected use only.

File: contrib/automation_tests/orbit_capture_loading.py (hunk 9 above)
Comment (akopich) [reply contrib/automation_tests/orbit_capture_loading.py:71]: Yeah, I've noticed this.  This time it makes the code fail.  I guess, I shouldn't expect F2 to rename it all.

File: contrib/automation_tests/test_cases/live_tab.py (hunk 10 above)
Comment (akopich) [reply contrib/automation_tests/test_cases/live_tab.py:26]: The term "scope" isn't introduced here. It's now heavily used in the codebase. But I'll add a comment.

File: contrib/automation_tests/test_cases/live_tab.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (dpallotti) [reply contrib/automation_tests/test_cases/live_tab.py:95]: I think it's fine to only check the prefix in this change, that's the type after all.

File: contrib/automation_tests/test_cases/symbols_tab.py [hunk 11]
```diff
@@ -453,10 +456,14 @@ def _execute(self):
         Capture().execute(self.suite)
 
         logging.info('Verifying function call counts')
-        VerifyFunctionCallCount(function_name='DrawFrame', min_calls=30,
-                                max_calls=3000).execute(self.suite)
-        VerifyFunctionCallCount(function_name='GgpIssueFrameToken', min_calls=30,
-                                max_calls=3000).execute(self.suite)
+        VerifyScopeTypeAndHitCount(function_name='DrawFrame',
```
Comment (dpallotti) [inline contrib/automation_tests/test_cases/symbols_tab.py:459]: `scope_name`

File: contrib/automation_tests/test_cases/symbols_tab.py [hunk 12]
```diff
@@ -453,10 +456,14 @@ def _execute(self):
...
+                                   scope_type="D",
+                                   min_hits=30,
+                                   max_hits=3000).execute(self.suite)
+        VerifyScopeTypeAndHitCount(function_name='GgpIssueFrameToken',
```
Comment (dpallotti) [inline contrib/automation_tests/test_cases/symbols_tab.py:463]: `scope_name`

**Review by dpallotti — APPROVED** [review]
Last parameter to rename, looks good otherwise.

File: contrib/automation_tests/orbit_capture_loading.py (hunk 9 above)
Comment (dpallotti) [reply contrib/automation_tests/orbit_capture_loading.py]: Depends on the IDE I guess. `PyCharm` worked generally fine for me on this.

File: contrib/automation_tests/test_cases/symbols_tab.py (hunk 11 above)
Comment (akopich) [reply contrib/automation_tests/test_cases/symbols_tab.py]: Thanks for shovelling these out.

**Review by danielfenner — APPROVED** [review]
lgtm

**Pull request merged**